### DISCUSSION TITLE
Search endpoint functionality

### DISCUSSION THREAD

**Discussion by Zan-Aslettco** [tidal-music/discussions#53] [General]
I'm playing around with the [search api](https://developer.tidal.com/apiref?spec=search&ref=search) and was wondering if there's an option for advanced searching criteria in the query field. Specifically, it would be cool if I could search for Albums with a specific search query for the title and the artist. One example is if I search for "Houses of the Holy Led Zeppelin", then the top result is this [karaoke album](https://tidal.com/browse/album/39417170). It would be nice if I could specify in my request that "Houses of the Holy" is a query on the album title and "Led Zeppelin" is a query on the artist.

**Comment by stefanmoro** [comment]
Hi @Zan-Aslettco,

Our Search team is currently working on an updated version of our search platform that will bring a few new features (we aim to expose this new version on TIDAL Developer when it's ready). I'll make sure to forward this message to the search team as well.

Thanks

**Reply by Zan-Aslettco** [reply]
Thanks for the response! I'm happy to answer any questions about my use case here to help the dev team.

**Comment by ungive** [comment]
You can always filter the results, and search whether one of the artists of the album has the name "Led Zeppelin". The first result is easily filtered out because the album artist is "The Karaoke Channel". The second album result is probably the one you are looking for ("Houses of the Holy (Remaster)" by "Led Zeppelin"). Doing some sort of client-side filtering also makes your application more robust against fluctuations in the order of API results and unpredictable "outlier" results like the one your experienced here.

I am also adding a hyphen between the track/album name and the artist name, just for good measure. It doesn't seem to make that big of a difference in which way you order it or if you add a hyphen or not, from my testing (even with short search queries like "Love August" for the tack "August" by "Love"), but adding a hyphen makes the track I am looking for appear a little earlier. Here's an example, where the number is the index in the "tracks" array, indicating which track exactly matches the search criteria:
```
Searching: August by Love
Love August 51
Love - August 48
August Love 51
August - Love 50
```

As you can see, the track I am actually looking for comes pretty late. It would be really nice if it were possible to specify artist/track title/album title separately, as the OP suggested! Otherwise a great API!

**Reply by Zan-Aslettco** [reply]
That's a great suggestion. This is what I ended up doing in the meantime. I still think some additional search parameters could make this API even better, but so far this has been working pretty well. I created a similarity algorithm to rank each result from the search API against the item I am looking for. I try to match up the album titles, the artist names, and the release year as well in order to figure out which result is most likely to be what I'm looking for.

One difficulty that I had to solve for was that album titles don't always match exactly nor are they unique. So the album I'm searching for might be "- (Subtract)" by Ed Sheeran, but on tidal the (Subtract) is omitted. A similar thing can happen with (Deluxe) qualifiers on album names. Or with artists like Weezer who have several albums entitled "Weezer" but are often referred to as the blue, red, teal, etc. album. The release date helped narrow the results down even further in cases like this.

**Comment by Kingpin2557** [comment]
I am confused with what api endpoint to use to search  album by title, I need the album id and track id + metadata in order to use later with the SDK. (I am doing a school assignment) my movie data gets fetched from TMDB and I will have the title dynamically but not the ID for these albums.
